Sources/Interaction/Style/InteractorStyleManipulator/index.js

```diff
@@ -421,12 +422,24 @@ function vtkInteractorStyleManipulator(publicAPI, model) {
     ) {
       model.currentManipulator.onButtonUp(model._interactor);
       model.currentManipulator.endInteraction();
-      model.currentManipulator = null;
-      model._interactor.cancelAnimation(publicAPI.onButtonDown);
+      if (
+        !(
+          button === MouseButton.LeftButton &&
+          model._interactor.isPointerLocked()
```

Comment:
It could be any Manipulator that triggered the pointerLocked mode. Let's not assume it is always on a LeftButton.

Maybe `endInteraction()` could return a boolean that tells if the manipulator accepts to be `ended`.

Reply:
Not sure to understand why you need to store a `canBeEnded` variable.

I let you propose your change and I'll review your code.